src/types/instanceOf-tests.ts

```diff
@@ -5,3 +5,12 @@ instanceOf(Error);
 
 // $ExpectType Decoder<TypeError, unknown>
 instanceOf(TypeError);
+
+// $ExpectType Decoder<RegExp, unknown>
+instanceOf(RegExp);
+
+// $ExpectType Decoder<Promise<string>, unknown>
+instanceOf<Promise<string>>(Promise);
```

Comment:
I originally just used this as an example of a class with a required argument but, because it is generic, the type that shows up on `$ExpectType` will depend on your TypeScript version.

It actually should be `Promise<unknown>` in TS 3.6 but for some reason dtslint is seeing it as `Promise<{}>`; so I gave up and just specified the argument here 🤔